modules/s3-bucket/main.tf

```diff
@@ -8,6 +8,52 @@ resource "aws_s3_bucket" "bucket" {
   )
 }
 
+resource "aws_s3_bucket_ownership_controls" "ownership" {
+  count = var.public_access ? 1 : 0
+  bucket = aws_s3_bucket.bucket.id
+  rule {
+    object_ownership = "BucketOwnerPreferred"
+  }
+}
+
+resource "aws_s3_bucket_public_access_block" "access_block" {
+  count = var.public_access ? 1 : 0
+  bucket = aws_s3_bucket.bucket.id
+
+  block_public_acls       = false
+  block_public_policy     = false
+  ignore_public_acls      = false
+  restrict_public_buckets = false
+}
+
+resource "aws_s3_bucket_acl" "bucket_acl" {
+  count = var.public_access ? 1 : 0
+  depends_on = [
+    aws_s3_bucket_ownership_controls.ownership,
+    aws_s3_bucket_public_access_block.access_block,
+  ]
+
+  bucket = aws_s3_bucket.bucket.id
+  acl    = "public-read"
+}
+
+resource "aws_s3_bucket_policy" "my_bucket_policy" {
+  count = var.public_access ? 1 : 0
+  bucket = aws_s3_bucket.bucket.bucket
+
+  policy = jsonencode({
+    Version = "2012-10-17",
```

Comment:
I am also curious here why you pick this date for version number?

Reply:
https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_policies_elements_version.html

> 2012-10-17. This is the current version of the policy language, and you should always include a Version element and set it to 2012-10-17. Otherwise, you cannot use features such as [policy variables](https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_policies_variables.html) that were introduced with this version.